apps/public-docsite-v9/src/Concepts/BuildTimeStyles.stories.mdx

```diff
@@ -0,0 +1,224 @@
+
+- [Quick start](?path=/docs/concepts-developer-quick-start--page)
+- [Styling components](?path=/story/concepts-developer-styling-components--page)
+
+While there is nothing wrong with the associated runtime costs of a CSS-in-JS engine, larger and more complex
+applications might want to optimize for performance. In Fluent UI, the expensive runtime work only
+happens on the first render of the component. This one-time work can be further optimized at build time by
+pre-computing and transforming styles. You can refer to the [last section](#what-is-being-optimized) of this page for more details
+about what the build time optimization does.
+
+## Setup build time style transforms
+
+- For application developers please use the [Webpack loader](#webpack)
+- For library developers please use the [Babel preset](#babel)
+
+### Webpack
+
+```sh
+yarn add @griffel/webpack-loader
+```
+
+In your Webpack config, simply add the `@griffel/webpack-loader` to the list of rules.
+
+```javascript
+// webpack.config.js
+module: {
+  rules: [
+    {
+      test: /\.(ts|tsx)$/,
+      exclude: /node_modules/,
+      use: {
+        loader: '@griffel/webpack-loader',
+      },
+    },
+    {
+      test: /\.(ts|tsx)$/,
+      exclude: /node_modules/,
+      use: {
+        loader: 'babel-loader',
+        options: {
+          babelOptions: {
+            presets: ['@babel/preset-typescript'],
+          },
+        },
+        // loader: 'ts-loader',
+      },
+    },
+  ];
+}
+```
+
+**We recommend adding the Griffel webpack loader after babel-loader or ts-loader.**. The underlying babel transforms
+used by the webpack loader are configured by default to use the most basic language features. Therefore to avoid
+extra configuration or parsing errors, using the webpack loader after transpilation is the recommended approach.
+
+#### Typescript (and additional transpilation) support
+
+The Webpack loader [evaluate styles at runtime using Linaria](#build-time-style-evaluation). For typescript
+it's necessary to install `@babel/preset-typescript` and configure it in the `babelOptions` of the loader. Similarly
+any other language features that might be required can also be configured here. While the order of loaders will
+ensure that the webpack loader will run on transpiled code (for a specific module), runtime evaluation can potentially require other
+modules that are not transpiled. Any additional language features can also be configured in the same way.
+
+### Babel
+
+The Babel preset is used internally by the Webpack loader. For applications the Webpack loader is appropriate
+since it is the most popular bundler.
+However, library developers might prefer to use the Babel preset since it is a better fit for tools such as [Rollup](https://rollupjs.org/guide/en/).
+
+```sh
+yarn add @griffel/babel-preset
+```
+
+Simply create a Babel configuration file such as `.babelrc` and add the preset.
```

Comment:
AFAIR [`.babelrc.js` is also a thing](https://babeljs.io/docs/en/configuration#javascript-configuration-files), I would rather use it to be able to use comments:

````
```js
module.exports = {
  presets: [
    [
      "@griffel",
      {
        // see https://github.com/microsoft/griffel/tree/main/packages/webpack-loader#configuring-babel-settings
        babelOptions: {
          presets: ["@babel/preset-typescript"]
        }
      }
    ]
  ]
};
```
````